src/ui/HorizontalScroll/HorizontalScroll.tsx

```diff
@@ -0,0 +1,117 @@
+import { FC, useRef, useCallback, useState, useEffect } from 'react';
```

Comment:
Have you checked framer-motion `AnimateSharedLayout`? I think all of these could have been replaced with just that. See the navbar of https://vasseneca.com/ or https://scr-dev.vercel.app/

Reply:
How does `framer-motion` fits in here? The scrolling aniamtion is just native browser aniamtion. I just add the 2 buttons to control the scroll which call the `element.scrollBy()` native function.

Reply:
I mean isn't a lot of logic is to make the fake scrollbar animation?  Like the `useTabAnimation`, `SelectedTabIndicator`, and the animation context.

Reply:
I imagine we can just hide the native scrollbar, add a div inside a tab and slap the "layoutId" on it. framer-motion will handle where that div is moved around and the animation as well.

Reply:
> I imagine we can just hide the native scrollbar, add a div inside a tab and slap the "layoutId" on it. framer-motion will handle where that div is moved around and the animation as well.

We are currently having a native scrollbar and the animation is handled without any extra effort.

> I mean isn't a lot of logic is to make the fake scrollbar animation? Like the useTabAnimation, SelectedTabIndicator, and the animation context.

These are for animating the moving of the `SelectedTabIndicator` (the color ribbon underneath the tab buttons indicating which tab is current selected) when the user switches tab (see gif). Those are not for scrolling.

![tab-animation](https://user-images.githubusercontent.com/54929186/144766614-fc734025-ee7b-4f00-bfd2-5b728e0fd0a0.gif)

However, I can look into using `framer-motion` for this animation as well

Reply:
> We are currently having a native scrollbar and the animation is handled without any extra effort.

There is a context to measure the button width and the scrollbar width to adjust them on different tab size right? And also the context is used to point where the scrollbar should go. Which framer-motion, just put a div inside Tab, and render that scrollbar on selected tab
```tsx
isSelected && <Scrollbar layoutId="scroll-bar" />
```
wrap the TabGroup inside `AnimatedShareLayout` and framer-motion should automatically handle the scrollbar.

Reply:
I got your point, will look into the `AnimatedShareLayout`.

But just to be clear, the blue ribbon in the gif is **not** the scrollbar. That's the `SelectedTabIndicator`

Reply:
> See the navbar of https://vasseneca.com/ or https://scr-dev.vercel.app/

I just don't know what to call it, but it is the same as the bar in those sites right?

Reply:
Yes

Reply:
Then the implementation can be simplify with framer-motion. Doesn't have to calculate the bar size on the fly. 

And don't  make the text bold on selected. It is shifting the other tabs.

Reply:
Made the switche to framer-motion